### ISSUE TITLE
Investigate bug in perfect foresight simulations in unstable

### ISSUE THREAD

**Issue by JohannesPfeifer** [DynareTeam/dynare#1124]
The user at http://www.dynare.org/phpBB3/viewtopic.php?f=1&t=7765 reports the following mod-file
```
%% CODE TO SIMULATE LOGISTIC MAP
%% DECLARATION OF ENDOGENOUS VARIABLES
var x;
parameters r s;

r=3.8;
s=0;

model;
x = r*x(-1)*(1 - x(-1)) + s*x(+1);
end;

initval;
x = 0.1;
end;

check;

%% DETERMINISTIC SIMULATION
simul(periods = 40, stack_solve_algo=0, maxit=100);
dsample 40;
rplot x;
```
With the unstable version, it crashes with 
```
Subscripted assignment dimension mismatch.

Error in sim1 (line 108)
            iA((1:length(v))+m,:) = [i_rows(r),i_cols_A1(c),v];
```
When I replace the parameter s in the model by 0, it runs as expected. But when I replace s by 0.1, there again is the same crash.

**Comment by MichelJuillard** [comment]
@JohannesPfeifer I fixed the dimension problem in my local copy of Dynare but the model doesn't converge. Are you sure that there is a solution to this model?

**Comment by JohannesPfeifer** [comment]
@MichelJuillard I guess there is no solution to the problem, but I am not sure. The main point of this issue was that the unstable crashed while the stable did not - while also not finding a solution.

**Comment by MichelJuillard** [comment]
Issue fixed by commit d60cc0a6faf2cc6660faead618c14c742ba782e9
